### ISSUE TITLE
Columns with CAST in the select statement are return as OBJECT data types in the resultSet

### ISSUE THREAD

**Issue by jadami10** [apache/pinot#11629]
We had a query that was doing `select long_column from X`. The user updated it to do `select long_column - Y from X`. This change the `columnDataTypes` for this to a `DOUBLE`. The user's assertions on column types broke, so they changed it to `SELECT CAST(long_column - Y AS LONG) from X`. But this also didn't work because `columnDataTypes` then showed `OBJECT`.

For `CAST`, the expectation is a successful result would show the casted data type

**Comment by Jackie-Jiang** [comment]
It this the complete query, or it is on top of an aggregation?
Currently we cannot tell the result type of `CAST` when it is an post aggregation, but the result type should be correct if it is modeled as a regular transform

**Comment by jadami10** [comment]
yes sorry, the actual query is more like `SELECT CAST(SUM(long_column) AS LONG)`. What is the limitation here? The type result of the cast should be unaffected by the aggregation, no?

**Comment by Jackie-Jiang** [comment]
That involves some implementation details. If the `CAST` is applied to a column (not aggregate function), it will be solved with `CastTransformFunction`, where the return type can be properly derived. If the `CAST` is applied to an aggregate function (we call it post-aggregation), it will be solved with scalar function (`DataTypeConversionFunctions.cast()`), where currently pinot cannot properly derive the result type
